application-glossary-machine-translation/application-glossary-machine-translation-api/src/main/java/org/xwiki/contrib/glossary/machineTranslation/internal/DefaultTranslationGlossaryManager.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.contrib.glossary.machineTranslation.internal;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.contrib.glossary.machineTranslation.TranslationGlossaryManager;
+import org.xwiki.contrib.translator.Translator;
+import org.xwiki.contrib.translator.TranslatorManager;
+import org.xwiki.contrib.translator.model.LocalePairs;
+import org.xwiki.contrib.translator.model.GlossaryUpdateEntry;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.ObjectReference;
+import org.xwiki.query.Query;
+import org.xwiki.query.QueryException;
+import org.xwiki.query.QueryManager;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+
+/**
+ * @version $Id$
+ */
+@Component
+@Singleton
+public class DefaultTranslationGlossaryManager implements TranslationGlossaryManager
+{
+    @Inject
+    private Logger logger;
+
+    @Inject
+    private Provider<XWikiContext> xwikiContextProvider;
+
+    @Inject
+    private TranslatorManager translatorManager;
+
+    @Inject
+    private QueryManager queryManager;
+
+    private Map<String, String> getLocalGlossaryEntries(Locale sourceLanguage, Locale targetLanguage)
+        throws QueryException
+    {
+        List<String[]> glossaryRawEntries = queryManager.createQuery("select sourceDoc.title, targetDoc.title "
+                    + "from XWikiDocument sourceDoc, XWikiDocument targetDoc, BaseObject glossaryObj "
+                    + "where glossaryObj.name = sourceDoc.fullName "
+                    + "and glossaryObj.name = targetDoc.fullName "
+                    + "and glossaryObj.className = 'Glossary.Code.GlossaryClass' "
```

Comment:
Do we need to restrict the scope to the "Glossary" space while glossaries can actually be created in any space?

Reply:
So, we concluded that for now we only support one glossary by wiki and following [this discussion](https://github.com/xwiki-contrib/application-glossary/pull/9#discussion_r1631356792) there are also some security implication.

Supporting multiple glossary could be a nice feature but for now we don't plan to support it.